grpc.go

```diff
@@ -524,6 +525,76 @@ func (a *agentGRPC) rollbackFailingContainerCreation(ctr *container) {
 	}
 }
 
+func changeToBundlePath(spec *specs.Spec) (string, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return cwd, err
+	}
+
+	rootfsPath := spec.Root.Path
+	if !filepath.IsAbs(rootfsPath) {
+		rootfsPath = filepath.Join(cwd, rootfsPath)
+	}
+
+	bundlePath := filepath.Dir(rootfsPath)
+
+	err = os.Chdir(bundlePath)
+	if err != nil {
+		return cwd, err
+	}
+
+	return cwd, nil
+}
+
+func writeSpecToFile(spec *specs.Spec) error {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	configPath := path.Join(cwd, "config.json")
+	f, err := os.OpenFile(configPath, os.O_RDWR|os.O_CREATE, 0666)
```

Comment:
Could you move the perms to a global `const` and make it an explicit `os.FileMode` value? Also, can't it be `0444`?